### PR TITLE
[TS] LPS-132373

### PR DESCRIPTION
**Pull request by joshuacords**
https://issues.liferay.com/browse/LPS-132373

Consider nonProxyHosts when applying configurations.

**Author:** @joshuacords

### CODE REVIEW

**Comment by joshuacords** [comment]
cc: @lipusz 
Hi Tibor, would you mind reviewing this?

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
Tests seem good!

**Comment by joshuacords** [comment]
And thanks Tibor for basically fixing this!

**Comment by lipusz** [comment]
ci:reevaluate:1097401_4827

**Comment by lipusz** [comment]
**Started reviewing...** :mag::nerd_face:

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ProxyConfig.java [hunk 1]
```diff
@@ -125,20 +125,24 @@ protected boolean hasHostAndPort() {
...
 			if (!_http.hasProxyConfig()) {
 				return false;
 			}
 
-			return Stream.of(
-				_networkHostAddresses
-			).allMatch(
-				host -> !_http.isNonProxyHost(host)
-			);
+			return isNotNonProxyHost();
```
[the remark this answers is not in the document]
Comment (joshuacords) [reply modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ProxyConfig.java:145]: If we just use "false" testShouldApplyConfigIfHttpHasProxyConfig Fails, but returning "true" they all pass. Going to think on this a little more.

[the remark this answers is not in the document]
Comment (joshuacords) [reply modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ProxyConfig.java:145]: It seems that the `shouldApplyConfig` and the `testShouldApplyConfigIfHttpHasProxyConfig` where specifically written to return true if (host && port) were not set and the host was not part of NonProxyHosts. But this doesn't make sense to me. From my understand of how proxies should work, we could simplify the method to simply:
```
protected boolean shouldApplyConfig() {
	if (hasHostAndPort()) {
		return Stream.of(
				      _networkHostAddresses
			      ).allMatch(
				      host -> !_http.isNonProxyHost(_http.getDomain(host))
			      );
	}

	return false;
}
```
And change the test. Thoughts Tibor?

[the remark this answers is not in the document]
Comment (lipusz) [reply modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/connection/ProxyConfig.java:145]: Yes, this is what I concluded to also.

**Review by lipusz — CHANGES_REQUESTED** [review]
Please update as we discussed. Thanks!

**Pull request closed without merging**